In [1]:
import pandas as pd

In [2]:
titanic_df = pd.read_csv("train.csv", delimiter=',', header=0)

In [3]:
titanic_df.tail()

PassengerId  Survived  Pclass                                      Name  \
886          887         0       2                     Montvila, Rev. Juozas   
887          888         1       1              Graham, Miss. Margaret Edith   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
889          890         1       1                     Behr, Mr. Karl Howell   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch      Ticket   Fare Cabin Embarked  
886    male  27.0      0      0      211536  13.00   NaN        S  
887  female  19.0      0      0      112053  30.00   B42        S  
888  female   NaN      1      2  W./C. 6607  23.45   NaN        S  
889    male  26.0      0      0      111369  30.00  C148        C  
890    male  32.0      0      0      370376   7.75   NaN        Q

## Condition Based Filtering

In [4]:
mask = titanic_df["Fare"] > 100
fare_100 = titanic_df[mask]

In [5]:
fare_100.shape

(53, 12)

In [6]:
female_df = fare_100[fare_100.Sex != "female"]

In [7]:
female_df.head()

PassengerId  Survived  Pclass                            Name   Sex  \
27            28         0       1  Fortune, Mr. Charles Alexander  male   
118          119         0       1        Baxter, Mr. Quigg Edmond  male   
305          306         1       1  Allison, Master. Hudson Trevor  male   
332          333         0       1       Graham, Mr. George Edward  male   
373          374         0       1             Ringhini, Mr. Sante  male   

       Age  SibSp  Parch    Ticket      Fare        Cabin Embarked  
27   19.00      3      2     19950  263.0000  C23 C25 C27        S  
118  24.00      0      1  PC 17558  247.5208      B58 B60        C  
305   0.92      1      2    113781  151.5500      C22 C26        S  
332  38.00      0      1  PC 17582  153.4625          C91        S  
373  22.00      0      0  PC 17760  135.6333          NaN        C

In [35]:
survive = titanic_df[
    (titanic_df["Sex"] == "female") & (titanic_df["Survived"] == 1)
]
survive.shape

(233, 12)

In [38]:
survive_male = titanic_df[
    (titanic_df["Sex"] == "male") & (titanic_df["Survived"] == 1)
]
survive_male.shape

(109, 12)

In [39]:
len(survive)

233

In [40]:
(233+109)/891

0.3838383838383838

In [41]:
survive_1stclass = titanic_df[
    (titanic_df["Pclass"] == 1) & (titanic_df["Survived"] == 1)
]
survive_1stclass.shape


(136, 12)

In [50]:
total1stclass = titanic_df[titanic_df["Pclass"] ==1]
total1stclass.shape

(216, 12)

In [47]:
len(survive_1stclass)/len(total1stclass) * 100

62.96296296296296

Out of 216 people 136 survived in the titaic who had first class tickets

## Filtering using existing methods

In [54]:
titanic_df[titanic_df["Cabin"].isna()]

PassengerId  Survived  Pclass                                      Name  \
0              1         0       3                   Braund, Mr. Owen Harris   
2              3         1       3                    Heikkinen, Miss. Laina   
4              5         0       3                  Allen, Mr. William Henry   
5              6         0       3                          Moran, Mr. James   
7              8         0       3            Palsson, Master. Gosta Leonard   
..           ...       ...     ...                                       ...   
884          885         0       3                    Sutehall, Mr. Henry Jr   
885          886         0       3      Rice, Mrs. William (Margaret Norton)   
886          887         0       2                     Montvila, Rev. Juozas   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch            Ticket     Fare Cabin Embarked  
0      male  22.0      1      0         A/5 21171   7.2500   NaN        S  
2    female  26.0      0      0  STON/O2. 3101282   7.9250   NaN        S  
4      male  35.0      0      0            373450   8.0500   NaN        S  
5      male   NaN      0      0            330877   8.4583   NaN        Q  
7      male   2.0      3      1            349909  21.0750   NaN        S  
..      ...   ...    ...    ...               ...      ...   ...      ...  
884    male  25.0      0      0   SOTON/OQ 392076   7.0500   NaN        S  
885  female  39.0      0      5            382652  29.1250   NaN        Q  
886    male  27.0      0      0            211536  13.0000   NaN        S  
888  female   NaN      1      2        W./C. 6607  23.4500   NaN        S  
890    male  32.0      0      0            370376   7.7500   NaN        Q  

[687 rows x 12 columns]

In [58]:
titanic_df[-titanic_df["Name"].duplicated()]

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

In [59]:
titanic_df[titanic_df["Ticket"].duplicated()]

PassengerId  Survived  Pclass  \
24            25         0       3   
71            72         0       3   
88            89         1       1   
117          118         0       2   
119          120         0       3   
..           ...       ...     ...   
876          877         0       3   
879          880         1       1   
880          881         1       2   
885          886         0       3   
888          889         0       3   

                                              Name     Sex   Age  SibSp  \
24                   Palsson, Miss. Torborg Danira  female   8.0      3   
71                      Goodwin, Miss. Lillian Amy  female  16.0      5   
88                      Fortune, Miss. Mabel Helen  female  23.0      3   
117                Turpin, Mr. William John Robert    male  29.0      1   
119              Andersson, Miss. Ellis Anna Maria  female   2.0      4   
..                                             ...     ...   ...    ...   
876                  Gustafsson, Mr. Alfred Ossian    male  20.0      0   
879  Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0      0   
880   Shelley, Mrs. William (Imanita Parrish Hall)  female  25.0      0   
885           Rice, Mrs. William (Margaret Norton)  female  39.0      0   
888       Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   

     Parch      Ticket      Fare        Cabin Embarked  
24       1      349909   21.0750          NaN        S  
71       2     CA 2144   46.9000          NaN        S  
88       2       19950  263.0000  C23 C25 C27        S  
117      0       11668   21.0000          NaN        S  
119      2      347082   31.2750          NaN        S  
..     ...         ...       ...          ...      ...  
876      0        7534    9.8458          NaN        S  
879      1       11767   83.1583          C50        C  
880      1      230433   26.0000          NaN        S  
885      5      382652   29.1250          NaN        Q  
888      2  W./C. 6607   23.4500          NaN        S  

[210 rows x 12 columns]

## Filter using Index

In [4]:
titanic_df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [6]:
titanic_df.loc[0,["Name","Age"]]

Name    Braund, Mr. Owen Harris
Age                        22.0
Name: 0, dtype: object

In [8]:
titanic_df.loc[:10:2, "Name"]

0                               Braund, Mr. Owen Harris
2                                Heikkinen, Miss. Laina
4                              Allen, Mr. William Henry
6                               McCarthy, Mr. Timothy J
8     Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)
10                      Sandstrom, Miss. Marguerite Rut
Name: Name, dtype: object

In [10]:
titanic_df.loc[:10:2]

PassengerId  Survived  Pclass  \
0             1         0       3   
2             3         1       3   
4             5         0       3   
6             7         0       1   
8             9         1       3   
10           11         1       3   

                                                 Name     Sex   Age  SibSp  \
0                             Braund, Mr. Owen Harris    male  22.0      1   
2                              Heikkinen, Miss. Laina  female  26.0      0   
4                            Allen, Mr. William Henry    male  35.0      0   
6                             McCarthy, Mr. Timothy J    male  54.0      0   
8   Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
10                    Sandstrom, Miss. Marguerite Rut  female   4.0      1   

    Parch            Ticket     Fare Cabin Embarked  
0       0         A/5 21171   7.2500   NaN        S  
2       0  STON/O2. 3101282   7.9250   NaN        S  
4       0            373450   8.0500   NaN        S  
6       0             17463  51.8625   E46        S  
8       2            347742  11.1333   NaN        S  
10      1           PP 9549  16.7000    G6        S

In [13]:
titanic_df.loc[:,"Cabin"]

0       NaN
1       C85
2       NaN
3      C123
4       NaN
       ... 
886     NaN
887     B42
888     NaN
889    C148
890     NaN
Name: Cabin, Length: 891, dtype: object

In [12]:
titanic_df.at[0, "Name"]

'Braund, Mr. Owen Harris'

In [5]:
temp = titanic_df.loc[45,]

In [6]:
titanic_df.where(titanic_df["Survived"] == 1, other=temp, axis=0 )

PassengerId Survived Pclass  \
0           NaN      NaN    NaN   
1             2        1      1   
2             3        1      3   
3             4        1      1   
4           NaN      NaN    NaN   
..          ...      ...    ...   
886         NaN      NaN    NaN   
887         888        1      1   
888         NaN      NaN    NaN   
889         890        1      1   
890         NaN      NaN    NaN   

                                                  Name     Sex   Age SibSp  \
0                                                  NaN     NaN   NaN   NaN   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0     1   
2                               Heikkinen, Miss. Laina  female  26.0     0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0     1   
4                                                  NaN     NaN   NaN   NaN   
..                                                 ...     ...   ...   ...   
886                                                NaN     NaN   NaN   NaN   
887                       Graham, Miss. Margaret Edith  female  19.0     0   
888                                                NaN     NaN   NaN   NaN   
889                              Behr, Mr. Karl Howell    male  26.0     0   
890                                                NaN     NaN   NaN   NaN   

    Parch            Ticket     Fare Cabin Embarked  
0     NaN               NaN      NaN   NaN      NaN  
1       0          PC 17599  71.2833   C85        C  
2       0  STON/O2. 3101282    7.925   NaN        S  
3       0            113803     53.1  C123        S  
4     NaN               NaN      NaN   NaN      NaN  
..    ...               ...      ...   ...      ...  
886   NaN               NaN      NaN   NaN      NaN  
887     0            112053     30.0   B42        S  
888   NaN               NaN      NaN   NaN      NaN  
889     0            111369     30.0  C148        C  
890   NaN               NaN      NaN   NaN      NaN  

[891 rows x 12 columns]